### PR TITLE
[rpc] remove dead mining code

### PR DESCRIPTION
**Pull request by jnewbery**
Removes dead code from after the generate method was removed

### CODE REVIEW

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -98,7 +99,7 @@ static UniValue getnetworkhashps(const JSONRPCRequest& request)
     return GetNetworkHashPS(!request.params[0].isNull() ? request.params[0].get_int() : 120, !request.params[1].isNull() ? request.params[1].get_int() : -1);
 }
 
-UniValue generateBlocks(std::shared_ptr<CReserveScript> coinbaseScript, int nGenerate, uint64_t nMaxTries, bool keepScript)
+static UniValue generateBlocks(CScript& coinbase_script, int nGenerate, uint64_t nMaxTries)
```
Comment (Empact) [inline src/rpc/mining.cpp]: nit: `coinbase_script` could be `const`

File: src/rpc/mining.cpp [hunk 2]
```diff
@@ -27,6 +26,8 @@
 #include <validationinterface.h>
 #include <versionbitsinfo.h>
 #include <warnings.h>
+#include <script/script.h>
+#include <univalue.h>
```
Comment (Empact) [inline src/rpc/mining.cpp]: nit: could sort includes

File: src/rpc/mining.cpp (hunk 1 above)
Comment (BelfordZ) [inline src/rpc/mining.cpp]: Just out of curiosity, how come the use of mixed casing? 

`coinbase_script`, `nMaxTries` all in the same function?

Comment (Empact) [reply src/rpc/mining.cpp]: > Variable (including function arguments) and namespace names are all lowercase, and may use _ to separate words (snake_case).
https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#coding-style-c

 > Patchsets should always be focused. For example, a pull request could add a feature, fix a bug, or refactor code; but not a mixture.
https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md

**Comment by practicalswift** [comment]
Concept ACK

Good catch!

File: src/rpc/mining.cpp (hunk 1 above)
Comment (jnewbery) [reply src/rpc/mining.cpp]: Changed. Thanks!

File: src/rpc/mining.cpp (hunk 2 above)
Comment (jnewbery) [reply src/rpc/mining.cpp]: Done

**Comment by jnewbery** [comment]
rebased

**Comment by MarcoFalke** [comment]
utACK 1b46a4889f894d63062454fb9f4c2e88689a9d75

**Comment by meshcollider** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/15748/commits/1b46a4889f894d63062454fb9f4c2e88689a9d75

**Comment by scravy** [comment]
utACK 1b46a48

**Comment by Empact** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/15748/commits/1b46a4889f894d63062454fb9f4c2e88689a9d75

**Pull request merged**